display_list/geometry/dl_region_unittests.cc

```diff
@@ -172,34 +378,54 @@ TEST(DisplayListRegion, TestAgainstSkRegion) {
     std::seed_seq seed{::testing::UnitTest::GetInstance()->random_seed()};
     std::mt19937 rng(seed);
 
-    SkRegion sk_region;
+    SkRegion sk_region1;
+    SkRegion sk_region2;
 
     std::uniform_int_distribution pos(0, 4000);
     std::uniform_int_distribution size(1, settings.max_size);
 
-    std::vector<SkIRect> rects_in;
+    std::vector<SkIRect> rects_in1;
+    std::vector<SkIRect> rects_in2;
 
     for (size_t i = 0; i < settings.iteration_count; ++i) {
       SkIRect rect =
           SkIRect::MakeXYWH(pos(rng), pos(rng), size(rng), size(rng));
-      rects_in.push_back(rect);
-      sk_region.op(rect, SkRegion::kUnion_Op);
+      rects_in1.push_back(rect);
+      sk_region1.op(rect, SkRegion::kUnion_Op);
+
+      rect = SkIRect::MakeXYWH(pos(rng), pos(rng), size(rng), size(rng));
+      rects_in2.push_back(rect);
+      sk_region2.op(rect, SkRegion::kUnion_Op);
     }
 
-    DlRegion region(std::move(rects_in));
+    DlRegion region1(rects_in1);
+    CheckEquality(region1, sk_region1);
 
-    // Do not deband the rectangles - identical to SkRegion::Iterator
-    auto rects = region.getRects(false);
+    DlRegion region2(rects_in2);
+    CheckEquality(region2, sk_region2);
 
-    std::vector<SkIRect> skia_rects;
+    auto intersects_1 = region1.intersects(region2);
+    auto intersects_2 = region2.intersects(region1);
+    auto sk_intesects = sk_region1.intersects(sk_region2);
+    EXPECT_EQ(intersects_1, intersects_2);
+    EXPECT_EQ(intersects_1, sk_intesects);
 
-    auto iterator = SkRegion::Iterator(sk_region);
-    while (!iterator.done()) {
-      skia_rects.push_back(iterator.rect());
-      iterator.next();
+    {
+      auto rects = region2.getRects(true);
+      for (const auto& r : rects) {
+        EXPECT_EQ(region1.intersects(r), sk_region1.intersects(r));
+      }
```

Comment:
Even more precisely, `sk_region1.contains(r)`?

Reply:
Oops, I misread. I thought it was checking for correctness of makeIntersection, but I see that is done via "CheckEquality" below.

Ignore this...